pkg/dogstatsd/packets/pool_manager.go

```diff
@@ -48,19 +51,30 @@ func (p *PoolManager) Put(x interface{}) {
 		return
 	}
 
+	var ref unsafe.Pointer
+	switch t := x.(type) {
+	case *[]byte:
+		ref = unsafe.Pointer(t)
+	case *Packet:
+		ref = unsafe.Pointer(t)
+	default:
+		log.Debugf("Unsupported type by pool manager")
+		return
+	}
+
```

Comment:
Do we need this ? The packet pool is heavily used by dsd and type inference is costly operation. I don't think it's worth is if the only goal is to detect unsupported type.

Reply:
I _think_ type assertion has very little overhead these days. It used to be quite expensive but not anymore: https://stackoverflow.com/questions/28024884/does-a-type-assertion-type-switch-have-bad-performance-is-slow-in-go/38245293

Reply:
Indeed, but I'm not sure we need it.